### ISSUE TITLE
Match subdirectories starting with `.` in `{dot: false}` mode

### ISSUE THREAD

**Issue by sgtlambda** [mrmlnc/fast-glob#226]
### Environment

  * OS Version: macOS 10.14.6
  * Node.js Version: 11.12.0

### Actual behavior
When the `dot` option is disabled (default behavior), `foo/{.,}**/{.,}*` matches `foo/bar/.file` but not `foo/.bar/file`

### Expected behavior
`foo/{.,}**/{.,}*` should still match `foo/.bar/file` even if the `dot` option is disabled because the pattern explicitly includes subdirectories starting with `.`.

### Steps to reproduce
Create demo files, run sample code

### Code sample

Directory structure:
```
.
└── foo
    ├── .bar
    │   └── file
    └── bar
        └── .file
```

```js
require('globby')([
    'foo/{.,}**/{.,}*'
]).then(console.log);
```

**Comment by mrmlnc** [comment]
You're right. This is a bug inside [`deep`](https://github.com/mrmlnc/fast-glob/blob/630f3e43701938f0f95f1a01b8c01379b1721092/src/providers/filters/deep.ts#L68-L70) filter.

**Comment by mrmlnc** [comment]
Will be fixed with `fast-glob@3.1.0`.
